plugins/kotlin/uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/internal/baseKotlinInternalUastUtils.kt

```diff
@@ -75,6 +75,9 @@ fun KtClassOrObject.toPsiType(): PsiType {
         PsiTypesUtil.getClassType(lightClass)
 }
 
+val KtElement.isFunctionWithImplicitUnitReturnType: Boolean
+    get() = this is KtNamedFunction && this.hasBlockBody() && !this.hasDeclaredReturnType()
```

Comment:
FYI, the counterpart is here: https://github.com/JetBrains/kotlin/blob/master/compiler/light-classes/src/org/jetbrains/kotlin/asJava/classes/ultraLightMembersCreator.kt#L231

Reply:
Even if a function has explicit `Unit` return type, the generated bytecode has a signature with `void` return type, so I believe even FE 1.0 LC modeling is wrong. I'll drop this part and just convert `Unit` to `void` for function return type. I'll also prepare how to test these, like actual compiler behavior v.s. LC behavior (and of course FIR LC), in general.